sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -528,7 +528,12 @@ private class BulkWriter(container: CosmosAsyncContainer,
         try {
           val itemOperation = resp.getOperation
           val itemOperationFound = activeBulkWriteOperations.remove(itemOperation)
-          assume(itemOperationFound) // can't find the item operation in list of active operations!
+          if (!itemOperationFound) {
```

Comment:
when the item not found, should we skip all the following processing? else the following check will fail:
 `     assume(totalScheduledMetrics.get() == totalSuccessfulIngestionMetrics.get)`

totalScheduledMetrics -> will only increase during scheduleInternal
totalSuccessfulIngestionMetrics -> will increase twice now (for original + retry operation)

Reply:
I do not think this is necessary - we would not re-schedule it - just add it again to the emitter?